Updated the rendering of the package links box

max-krichenbauer opened the pull request:
This is to address #91 and #92.

![image](https://user-images.githubusercontent.com/15894344/72589881-cfc8fd00-393f-11ea-85a8-08ea032a9b81.png)

- RosEco link now has a text
- All links now have tool-tips
- Wiki and RosEco links are now enabled

Caveats:
- It was suggested that wiki links are not displayed
  for ROS2 packages, but I believe some ROS2 packages
  do indeed have wiki pages. In the long run,
  I heard that the wiki should be discontinued anyways,
  so I'm not sure how much sense it makes to add
  specific code to detect if a package is ROS1 or ROS2
- The wiki link is supposed to be disabled when no
  wiki page for the package exists. But no such
  code is in the rosindex sources yet and I don't know
  of any way to detect it except querying the page
  and parsing it's contents for "This page does not exists".
- On the RosEco link there was a conditional to make
  it disabled when no documentation for the package
  was available on docs.ros.org. I don't understand
  how RosEco works, but it seemed like a copy-and-paste
  mistake, so I removed it.

max-krichenbauer committed:
Updated the rendering of the package links box

This is to address #91 and #92.
- RosEco link now has a text
- All links now have tool-tips
- Wiki and RosEco links are now enabled

Caveats:
- It was suggested that wiki links are not displayed
  for ROS2 packages, but I believe some ROS2 packages
  do indeed have wiki pages. In the long run,
  I heard that the wiki should be discontinued anyways,
  so I'm not sure how much sense it makes to add
  specific code to detect if a package is ROS1 or ROS2
- The wiki link is supposed to be disabled when no
  wiki page for the package exists. But no such
  code is in the rosindex sources yet and I don't know
  of any way to detect it except querying the page
  and parsing it's contents for "This page does not exists".
- On the RosEco link there was a conditional to make
  it disabled when no documentation for the package
  was available on docs.ros.org. I don't understand
  how RosEco works, but it seemed like a copy-and-paste
  mistake, so I removed it.

tfoote commented:
For the wiki link we really don't want ROS 2 content on the wiki. Can we conditionally add that link just based on ROS 1 vs ROS 2, maybe using the rosdistro lists as a flag whether to link to it or not.

For the roseco links. I believe it uses the yaml files generated by the documentation job to generate it's rendering thus it has no data if there's no documentation job so I believe that wasn't a mistake.

max-krichenbauer committed:
Updated the link box based on feedback.

Wiki link now only displayed if the package has a ROS1 distro.
RosEco link only enabled if package is documented.

max-krichenbauer commented:
Thank you @tfoote 
I updated the PR to address both issues:
- now the wiki link is only displayed if the package has a ROS1 distro in the list
- ROSeco link disabled if package isn't documented

tfoote commented on _includes/package_links.html:
```diff
@@ -13,9 +13,50 @@
+       {%if hide_link_labels%}<br>{%endif%}
+       Browse Code
+  </a>
+  {% if page.available_distros["noetic"]
```
This list is available in the site variables. https://github.com/ros-infrastructure/rosindex/blob/ros2/_config.yml#L94

I think you can get the list as `site.ros_distros + site.old_ros_distros` instead of hard coding them here. 

Does liquid handle key errors appropriately on a dict for an if? It's usually a bad idea to test for keys in an if statement. I usually want to test 'foo' in page.available_distros  or something more like `if set(site.old_ros_distros + site.ros_distros) & set(page.available_distros.keys())`

max-krichenbauer replied:
Thanks. I switched to using `ros_distro` and `old_ros_distro` instead.

As for handling missing keys, using liquid `if` to test for a keys was used in other parts of the code. If that's a problem I can search for some different approaches.

tfoote replied:
Thanks, the missed keys I was thinking python. But this is ruby so it's slightly different and if the pattern works elsewhere hopefully it works here too.

tfoote commented on _includes/package_links.html:
```diff
@@ -13,9 +13,50 @@
+  {% if page.available_distros["noetic"]
+     or page.available_distros["melodic"]
+     or page.available_distros["kinetic"]
+     or page.available_distros["ardent"]
```
Note that ardent and bouncy are old ros2 distros this is why we don't want things duplicated in the code.

max-krichenbauer replied:
Understood. I switched it to using the site's config variables `ros_distro` and `old_ros_distro` instead.

max-krichenbauer committed:
Swiched to using site config for ROS1 distro lookup.
